src/Sarif/Schemata/Sarif.schema.json

```diff
@@ -146,1050 +156,1072 @@
-      "additionalProperties": false,
-      "type": "object",
-      "properties": {
-
-        "uri": {
-          "description": "The location of the file as a valid URI.",
-          "type": "string",
-          "format": "uri"
         },
-
-        "uriBaseId": {
-          "description": "A string that identifies the conceptual base for the 'uri' property (if it is relative), e.g.,'$(SolutionDir)' or '%SRCROOT%'.",
-          "type": "string"
-        },
-
-        "region": {
-          "description": "The region within the file where the result was detected.",
-          "$ref": "#/definitions/region"
-        }
+        "required": [ "message" ]
       },
+      "physicalLocation": {
+        "description": "A physical location relevant to a result. Specifies a reference to a programming artifact together with a range of bytes or characters within that artifact.",
+        "additionalProperties": false,
+        "type": "object",
+        "properties": {
 
-      "required": [ "uri" ],
-      "dependencies": {
-        "uriBaseId": [ "uri" ]
-      }
-    },
-    "region": {
-      "description": "A region within a file where a result was detected.",
-      "additionalProperties": false,
-      "type": "object",
-      "properties": {
-
-        "startLine": {
-          "description": "The line number of the first character in the region.",
-          "type": "integer",
-          "minimum": 1
-        },
-
-        "startColumn": {
-          "description": "The column number of the first character in the region.",
-          "type": "integer",
-          "minimum": 1
-        },
-
-        "endLine": {
-          "description": "The line number of the last character in the region.",
-          "type": "integer",
-          "minimum": 1
-        },
+          "uri": {
+            "description": "The location of the file as a valid URI.",
+            "type": "string",
+            "format": "uri"
+          },
 
-        "endColumn": {
-          "description": "The column number of the last character in the region.",
-          "type": "integer",
-          "minimum": 1
-        },
+          "uriBaseId": {
+            "description": "A string that identifies the conceptual base for the 'uri' property (if it is relative), e.g.,'$(SolutionDir)' or '%SRCROOT%'.",
+            "type": "string"
+          },
 
-        "offset": {
-          "description": "The zero-based offset from the beginning of the file of the first byte or character in the region.",
-          "type": "integer",
-          "minimum": 0
+          "region": {
+            "description": "The region within the file where the result was detected.",
+            "$ref": "#/definitions/region"
+          }
         },
 
-        "length": {
-          "description": "The length of the region in bytes or characters.",
-          "type": "integer",
-          "minimum": 0
+        "required": [ "uri" ],
```

Comment:
`uri` can no longer be a required property of `physicalLocation`, since there is a circumstance where it's not required. The smart validator will need to handle this.